### PR TITLE
Must gather - first test

### PR DESCRIPTION
**Pull request by AviadP**

### CODE REVIEW

File: tests/manage/cluster/test_must_gather.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+from ocs_ci.framework.testlib import ManageTest, tier4
+from ocs_ci.ocs import openshift_ops
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.utils import collect_ocs_logs
+
+logger = logging.getLogger(__name__)
+
+
+@tier4
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: tier1

File: tests/manage/cluster/test_must_gather.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        init OCP() object
+        """
+
+        self.ocs = openshift_ops.OCP()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = pod.get_all_pods(selector=['app=must-gather'])
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: need to verify that the must gather pod is deleted

File: tests/manage/cluster/test_must_gather.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        pods = self.get_ocs_pods()
+        logger.info(f"Logs: {logs}")
+        logger.info(f"pods list: {pods}")
+        assert set(sorted(logs)) == set(sorted(pods))
+
+        # Verify logs file are not empty
+
+    def make_directory(self):
+        """
+
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: delete empty line

File: tests/manage/cluster/test_must_gather.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            must_gather_pods = pod.get_all_pods(selector=['app=must-gather'])
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+            if must_gather_pods:
+                for pod_to_del in must_gather_pods:
+                    self.ocp_obj.wait_for_delete(resource_name=pod_to_del)
+
+        request.addfinalizer(finalizer)
+
+    def test_must_gather(self):
+
```
Comment (RazTamir) [inline tests/manage/cluster/test_must_gather.py]: Please add a description to the test

File: tests/manage/cluster/test_must_gather.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        # Compare running pods list to "/pods" subdirectories
+        logger.info("Checking logs tree")
+        logs = self.get_log_directories(directory)
+        pods = self.get_ocs_pods()
+        logger.info(f"Logs: {logs}")
+        logger.info(f"pods list: {pods}")
+        assert set(sorted(logs)) == set(sorted(pods))
+
+        # Verify logs file are not empty
```
Comment (RazTamir) [inline tests/manage/cluster/test_must_gather.py]: where is that verification?

File: tests/manage/cluster/test_must_gather.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    def get_ocs_pods(self):
+        """
+        Get list of openshift-storage pods
+
+        Returns:
+            list: pods in openshift-storage namespace
+
+        """
+
+        pods = (self.ocs.get_pods(namespace='openshift-storage'))
```
Comment (RazTamir) [inline tests/manage/cluster/test_must_gather.py]: redundant parenthesis

File: tests/manage/cluster/test_must_gather.py (hunk 5 above)
Comment (AviadP) [reply tests/manage/cluster/test_must_gather.py]: currently only 1st test is ready. this one would be next

File: tests/manage/cluster/test_must_gather.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        collect_ocs_logs(dir_name=directory, ocp=False)
+        logger.info("Collecting logs - Done!")
+
+        # Compare running pods list to "/pods" subdirectories
+        logger.info("Checking logs tree")
+        logs = self.get_log_directories(directory)
+        pods = self.get_ocs_pods()
+        logger.info(f"Logs: {logs}")
+        logger.info(f"pods list: {pods}")
+        assert set(sorted(logs)) == set(sorted(pods))
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: please add an assertion message

**Comment by AviadP** [comment]
closing it, added new one with 2nd test aswell

**Pull request closed without merging**